source/AI.cpp

```diff
@@ -3285,21 +3285,22 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
-			
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Boarding target priority");
 			for(const shared_ptr<Ship> &other : ships)
 				if(CanBoard(ship, *other))
 				{
 					if(shift && !other->IsYours())
 						continue;
 					
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double b = distancePriority ? other->Position().Distance(ship.Position()) : -other->Cost();
```

Comment:
If there are two potential ships with the same Cost(), then I would still expect the closest one to be selected. And if the most valuable ship is very far away, then I would also expect a high value closer-by ship to be selected (even if the algorithm optimizes for cost).

Tehhowch seems to have made [a very similar remark on the original request](https://github.com/endless-sky/endless-sky/issues/1229#issuecomment-347930252).

Reply:
When thinking about this a bit more; If I board based on value/costs, then I would want to board the most valuable ship that is relatively close to me. If I wanted to board a ship that is far away from me, then I would first fly some distance towards that ship (away from the ships I don't want to board) before pressing boarding.

Reply:
you can't see all of the ships in the system, even at max scrolling you don't see much so how would you know there is a ship far away costier ? 
I agree with your point that if there is a ship closer of the same price, you should choose it, and actually even that the "costier" option should be changed to "best" then and be based on cost/distance ratio, I mostly intended this to avoid boarding useless fighters when there is good stuff but their price would be so low anyway that it would work still

Reply:
I found the perfect thing to divide by at least according to my standards : 200, it means that for example if a firebird is relatively close but a carrier is shown on screen, the carrier will be selected, and escorts will pretty much almost be ignored  except if there isn't anything else on screen and even further

Reply:
I'm happy that both distance and costs are now considered. Unfortunately I don't have time to test the fine-tuning, but maybe somebody else wants to look into that.

Reply:
yeah anyone can do it not just a coder all that is needed is to be able to compile this and say if this is too much of a focus on price or distance